packages/engine/src/avatar/AvatarInputSystem.ts

```diff
@@ -144,11 +160,61 @@ const onKeyP = () => {
   getMutableState(RendererState).debugEnable.set(!getMutableState(RendererState).debugEnable.value)
 }
 
+const isAvatarClicked = () => {
+  const hits = Physics.castRayFromCamera(
+    Engine.instance.camera,
+    Engine.instance.pointerState.position,
+    Engine.instance.physicsWorld,
+    raycastComponentData
+  )
+  if (hits.length) {
+    const hit = hits[0]
+    const hitEntity = (hit.body?.userData as any)?.entity as Entity
+    if (typeof hitEntity !== 'undefined' && hitEntity == Engine.instance.localClientEntity) {
+      return true
+    }
+  }
+  return false
+}
+
+let primaryClickCount = 0
+const primaryClickTimeout = 2
+let primaryClickTimer = 0
+const reDoubleClickTimeout = 0.2
+let reDoubleClickTimer = 0
```

Comment:
we should name this `doubleClickTimer` and `doubleClickTimeout`

Reply:
yup renamed the variables as well

Reply:
i think we should make a function in client system , which allows us to detect double tap for any  button in the system

Reply:
we would just have to paste this function